src/server/game/Spells/SpellEffects.cpp

```diff
@@ -3020,6 +3020,13 @@ void Spell::EffectSummonPet(SpellEffIndex effIndex)
             //OldSummon->Relocate(px, py, pz, OldSummon->GetOrientation());
             //OldSummon->SetMap(owner->GetMap());
             //owner->GetMap()->Add(OldSummon->ToCreature());
+            if (OldSummon->getPetType() == SUMMON_PET)
+            {
+                OldSummon->SetHealth(OldSummon->GetMaxHealth());
+                OldSummon->SetPower(POWER_MANA, OldSummon->GetMaxPower(POWER_MANA));
+                OldSummon->SetPower(POWER_ENERGY, OldSummon->GetMaxPower(POWER_ENERGY));
```

Comment:
Blizzlike way for this:
```
OldSummon->SetPower(POWER_MANA, OldSummon->GetMaxPower(POWER_MANA));
OldSummon->SetPower(POWER_ENERGY, OldSummon->GetMaxPower(POWER_ENERGY));
```
Should look like this, just in case (if player has different power type):

`OldSummon->SetPower(OldSummon->getPowerType(), OldSummon->GetMaxPower(OldSummon->getPowerType()));`